packages/gatsby-theme-project-portal/src/components/SiteSearch.tsx

```diff
@@ -15,17 +15,26 @@ export const SiteSearch: FunctionComponent<SearchProps> = ({
 }: SearchProps) => {
   const [searchQuery, setSearchQuery] = useState([])
   const [queryResults, setQueryResults] = useState([])
+  const [errorMessage, setErrorMessage] = useState("")
 
   useEffect(() => {
     if (searchQuery.length > 0) {
-      let searchResults = index.search(searchQuery)
-      let results = []
-      searchResults.forEach(function (result) {
-        results.push(db[result.ref])
-      })
-      setQueryResults(searchResults)
+      // catch the error, and display an error message to help user
+      try {
+        let searchResults = index.search(searchQuery)
+        setQueryResults(searchResults)
+        setErrorMessage("")
+      } catch (error) {
+        if (error instanceof lunr.QueryParseError) {
+          setErrorMessage(error.message)
+          return
```

Comment:
Is this the only kind of error which lunr can throw? Can we get it to throw other errors?

Reply:
Should we also handle the "other" kind of errors? Or it would cause the search to crash just like the original error right? We don't have to display any message but if we have `throw error` and dont catch/handle it then it would cause `SiteSearch` to crash i think?